Mathlib/Tactic/Peel.lean

```diff
@@ -223,19 +223,19 @@ def peelArgsIff (l : List Name) : TacticM Unit := withMainContext do
       peelArgsIff hs
 
 elab_rules : tactic
-  | `(tactic| peel $[$num?:num]? $e:term $[with $n? $l?*]?) => withMainContext do
+  | `(tactic| peel $[$num?:num]? $e:term $[with $l?* $n]?) => withMainContext do
```

Comment:
Could you leave the name as `n?` as a reminder it's an Option?

Reply:
@kmill it *shouldn't* be an option when `with` is present. This argument should be mandatory in that case. Also, I think that would make parsing impossible. How would you know whether the final name is the last variable name (in the case when this argument is not present) or the name of the peeled hypothesis (when it is present)?

Reply:
The question mark is just part of the name, it doesn't affect the syntax pattern. I like using ? To mark when syntax pieces have an Option type. Correct me if I'm wrong about the type here.